### PR TITLE
Avoid importing the whole of `react-icons` (if treeshaking is broken)

### PR DESCRIPTION
**Pull request by fregante**
### Reasons for making this change

Due to the `bootstrap-4` package, my bundles end up with large chunks of `react-icons` package, often being the largest import:

Screen Shot 11

This is a common issue with `react-icons` and apparently the solution is avoid the main package and use `@react-icons/all-files` instead:

- https://github.com/react-icons/react-icons/issues/154

The main package is set up for tree-shaking but it's not working for me, even if my bundler correctly tree-shakes `lodash-es` for example

### Checklist

* [ ] **I'm updating documentation**
  - [ ] I've [checked the rendering](https://react-jsonschema-form.readthedocs.io/en/latest/#contributing) of the Markdown text I've added
* [ ] **I'm adding or updating code**
  - [x] I've added and/or updated tests. I've run `npm run test:update` to update snapshots, if needed.
  - [ ] I've updated [docs](https://react-jsonschema-form.readthedocs.io/) if needed
  - [ ] I've updated the [changelog](https://github.com/rjsf-team/react-jsonschema-form/blob/master/CHANGELOG.md) with a description of the PR
* [ ] **I'm adding a new feature**
  - [ ] I've updated the playground with an example use of the feature

### CODE REVIEW

**Comment by heath-freenome** [comment]
@fregante FYI, the v4 branch is locked and soon the `rjsf-v5` branch will be landing in `master`. If you want a head-start on the changes, you could rebase this PR on-top-of that branch, or wait for it to be merged to `master` before continuing

**Comment by heath-freenome** [comment]
@fregante `master` now contains the beta release, feel free to rebase and finish your PR... Do add tests please

**Comment by fregante** [comment]
Tests for what? This only affects builds. To test this one would have to set up a webpack build and  set a size limit.

**Comment by heath-freenome** [comment]
> Tests for what? This only affects builds. To test this one would have to set up a webpack build and set a size limit.

ah, right... well as long as the build still runs then you are ok. You will need to rebase and resolve conflicts

**Comment by fregante** [comment]
It should be good to go now, but the lockfile has conflicts again, so maybe it's best to resolve them and merge the PR when you're ready, otherwise I have to resolve them daily.

Also the `postinstall` script (lerna bootstrap) is regularly failing for me on Codespaces.

**Comment by heath-freenome** [comment]
@fregante I can't update your PR since it is on your fork... I can however upload the `package-lock.json` I get when I did the `npm install` with your `package.json` change... Try replacing what you have with this and I'll rerun the build:

[package-lock.zip](https://github.com/rjsf-team/react-jsonschema-form/files/9457947/package-lock.zip)

**Comment by heath-freenome** [comment]
> Also the `postinstall` script (lerna bootstrap) is regularly failing for me on Codespaces.

Are you running node 16? I do know that `lerna bootstrap` can take a long time and once crashed my computer.

**Comment by fregante** [comment]
> @fregante I can't update your PR since it is on your fork..

You should be able to push any changes:
Screen Shot

This is the issue I get:

```

@fregante ➜ /workspaces/react-jsonschema-form (patch-1) $ npm install
npm WARN deprecated source-map-resolve@0.6.0: See https://github.com/lydell/source-map-resolve#deprecated

> react-jsonschema-form@5.0.0 postinstall
> lerna bootstrap

lerna notice cli v5.4.3
lerna info Bootstrapping 11 packages
lerna info Installing external dependencies
lerna ERR! npm install exited undefined in '@rjsf/playground'
lerna ERR! npm install exited undefined in '@rjsf/playground'
lerna WARN complete Waiting for 3 child processes to exit. CTRL-C to exit immediately.
npm ERR! code 1
npm ERR! path /workspaces/react-jsonschema-form
npm ERR! command failed
npm ERR! command sh -c lerna bootstrap

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/codespace/.npm/_logs/2022-08-31T02_58_01_174Z-debug.log
```

I tried again by checking out the lockfile from master and running `npm install` again, even if the postinstall failed. Let's see

**Comment by fregante** [comment]
> Are you running node 16?

```js
@fregante ➜ /workspaces/react-jsonschema-form (patch-1) $ npm --version
node --version
7.24.2
@fregante ➜ /workspaces/react-jsonschema-form (patch-1) $ node --version
v16.14.2
```

**Comment by fregante** [comment]
I'll upload your lockfile

**Comment by heath-freenome** [comment]
Also, if this passes, can you update the `CHANGELOG.md` file with a description of your fix and the issue it closes?

File: CHANGELOG.md [hunk 1]
```diff
@@ -37,6 +37,9 @@ should change the heading of the (upcoming) version to include a major version b
 - Upgraded from the `1.x` to `2.x` version of `semantic-ui-react`
 - Added support for `schema.examples` in the material ui theme fixing (https://github.com/rjsf-team/react-jsonschema-form/issues/2368, https://github.com/rjsf-team/react-jsonschema-form/issues/2557)
 
+## @rjsf/bootstrap-4
+- Avoid importing the whole of `react-icons` (https://github.com/rjsf-team/react-jsonschema-form/pull/3046)
```
Comment (fregante) [inline CHANGELOG.md:41]: If you prefer:

```suggestion
- Avoid importing the whole of `react-icons` (https://github.com/rjsf-team/react-jsonschema-form/pull/3046, https://github.com/react-icons/react-icons/issues/154)
```

Comment (heath-freenome) [reply CHANGELOG.md:41]: @fregante Yes, that suggestion works, go ahead and commit it along with the updates to the snapshots with the new icons.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -11,7 +11,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        node-version: [14.x, 16.x, 18.x]
```
[the remark this answers is not in the document]
Comment (fregante) [reply .github/workflows/ci.yml]: Bad merge, it was already reverted

**Comment by fregante** [comment]
Sorry I cannot make any changes, I even tried it on my computer (macOS) and `npm install` still fails. Feel free to commit any missing changes and push them to my repo, it's possible.

**Comment by heath-freenome** [comment]
I still can't make changes so here are the updated snapshots
[snapshots.zip](https://github.com/rjsf-team/react-jsonschema-form/files/9465625/snapshots.zip)

**Comment by fregante** [comment]
> I still can't make changes

You're definitely allowed to considering you just pushed 33f4115 to my repo 😁 

https://github.com/fregante/react-jsonschema-form/commits/patch-1


See

https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/committing-changes-to-a-pull-request-branch-created-from-a-fork



> I still can't make changes so here are the updated snapshots
> [snapshots.zip](https://github.com/rjsf-team/react-jsonschema-form/files/9465625/snapshots.zip)

Thank you! Committed

**Review by heath-freenome — APPROVED** [review]

**Comment by heath-freenome** [comment]
@fregante Go ahead and merge, unless you'd rather I do it

**Comment by heath-freenome** [comment]
done!

**Comment by fregante** [comment]
# 🙌

**Pull request merged**